### PR TITLE
BS-4982: Recreated mime2extension to use map

### PR DESCRIPTION
**Pull request by muhammadharis**
https://jira.enflick.com/browse/BS-4982

O(1) lookup.

Mime to extension isn't unique, just handled it the way the original library assigned priority.

### CODE REVIEW

File: mime_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	err, mime = Lookup(".TXT")
+	makeAssertion(t, err, nil, mime, "text/plain")
+
+	err, mime = Lookup("htm")
+	makeAssertion(t, err, nil, mime, "text/html")
+
+	err, mime = Lookup("foo")
+	makeAssertion(t, err, errors.New("Not found"), mime, "")
+
+	fmt.Println(Extensions("video/mp4"))  // => [mp4 mp4v mpg4]
```
Comment (stephenhyde) [inline mime_test.go]: were these meant to be assertions?

File: list.go [hunk 2]
```diff
@@ -1,993 +1,801 @@
...
-		MimeExtension{"zaz", "application/vnd.zzazz.deck+xml"},
-		MimeExtension{"zip", "application/zip"},
-		MimeExtension{"zir", "application/vnd.zul"},
-		MimeExtension{"zirz", "application/vnd.zul"},
-		MimeExtension{"zmm", "application/vnd.handheld-entertainment+xml"} }
 }
+
+func createExtToMime(mimeToExt map[string][]string) (map[string]string) {
+	extToMime := map[string]string {
+	}
```
Comment (stephenhyde) [inline list.go]: make one line

File: mime_test.go (hunk 1 above)
Comment (muhammadharis) [reply mime_test.go]: oops, must've left these in after i finished converting them to unit tests

**Review by stephenhyde — COMMENTED** [review]
Can you remove the parenthesis around map for functions createExtToMime and createMimeToExt. Not part of this pr but ocd on my end.

File: list.go [hunk 3]
```diff
@@ -1,993 +1,800 @@
...
+			extToMime[ext] = k
+		}
+	}
+	return extToMime
+}
+
+func createMimeToExt() (map[string][]string) {
+	mimeToExt := map[string][]string {
+		 "application/vnd.lotus-1-2-3": []string{"123"},
+		 "text/vnd.in3d.3dml": []string{"3dml"},
```
Comment (stephenhyde) [inline list.go]: you can omit []string from each one of these

File: mime2extension.go [hunk 4]
```diff
@@ -1,42 +1,34 @@
...
-
-	for _, item := range *List_items {
-		if item.Extension == extension {
-			return nil, item.Mime
-		}
+	if val, ok := bdMap.extToMime[extension]; ok {
+		return nil, val
 	}
 
 	return errors.New("Not found"), ""
```
Comment (stephenhyde) [inline mime2extension.go]: errors should be lowercase ( I know it was here before you lol)

File: mime2extension.go [hunk 5]
```diff
@@ -46,32 +38,26 @@ func Lookup( file_path string ) (error, string) {
...
 
-	for _, item := range *List_items {
-		if item.Mime == mime {
-			return nil, item.Extension
-		}
+	if val, ok := bdMap.mimeToExt[mime]; ok {
+		return nil, val[0]
 	}
-
 	return errors.New("Not found"), ""
```
Comment (stephenhyde) [inline mime2extension.go]: same

**Review by stephenhyde — APPROVED** [review]

**Pull request merged**